### PR TITLE
Fix StackOverflowException in Circuit Breaker DI container setup

### PR DESCRIPTION
**Pull request by DaveAurionix**
Addresses issue #17

### CODE REVIEW

**Comment by DaveAurionix** [comment]
I'd like to merge as soon as any two reviewers have approved it unless anyone raises an objection.  This isn't a blocking/urgent issue for us though.

**Review by ilivewithian — APPROVED** [review]

**Review by AndyRB — APPROVED** [review]

**Comment by DaveAurionix** [comment]
Sorry, busy day. @ilivewithian thanks for chasing and merging.

**Pull request merged**
